custom_components/localtuya/fan.py

```diff
@@ -19,23 +19,25 @@
 import voluptuous as vol
 
 from homeassistant.components.fan import (
+    FanEntity,
+    PLATFORM_SCHEMA,
     SPEED_LOW,
     SPEED_MEDIUM,
     SPEED_HIGH,
-    FanEntity,
     SUPPORT_SET_SPEED,
     SUPPORT_OSCILLATE,
     SUPPORT_DIRECTION,
-    PLATFORM_SCHEMA,
 )
 from homeassistant.const import CONF_ID, CONF_FRIENDLY_NAME, STATE_OFF
 import homeassistant.helpers.config_validation as cv
 
 from . import BASE_PLATFORM_SCHEMA, prepare_setup_entities, import_from_yaml
-from .pytuya import FanDevice
+from .pytuya import TuyaDevice
 
 _LOGGER = logging.getLogger(__name__)
 
+PLATFORM = "fan"

```

Comment:
This is good but we should import from home Assistant instead (at top):

```python
from homeassistant.components.fan import DOMAIN
```
Same for the other platforms.

Reply:
Great, I knew it probably was already defined somewhere. 👍